src/features/machines/swapIntentMachine.ts

```diff
@@ -501,14 +502,37 @@ export const swapIntentMachine = setup({
           ],
         },
 
-        onDone: {
-          target: "Completed",
-
-          actions: {
-            type: "setIntentHash",
-            params: ({ event }) => event.output,
+        onDone: [
+          {
+            target: "Completed",
+            guard: {
+              type: "isOk",
+              params: ({ event }) => event.output,
+            },
+            actions: {
+              type: "setIntentHash",
+              params: ({ event }) => event.output.value as string,
```

Comment:
It's better to assert value instead of typecasting, because guard could be removed for some reason:
```
assert(event.output.tag === 'ok')
return event.output.value
```